### PR TITLE
Add diffraction export to core (and fix elastic logging and add profiling endpoints)

### PR DESCRIPTION
**Pull request by buggtb**
This came from the old repo and was required to add diffraction export to the exporter.

A couple of extra things have gone in here, some updates to elastic logging and also adding the profiler)

### CODE REVIEW

File: core/export/exporter.go [hunk 1]
```diff
@@ -235,6 +241,42 @@ func (m *Exporter) MakeExportFilesZip(svcs *services.APIServices, outfileNamePre
 			return nil, err
 		}
 	}
+	if wantDiffractionPeak {
+		fmt.Println(files.diffraction.Title)
+		fmt.Printf("%v Locations with Peaks found\n", len(files.diffraction.Locations))
+		csv, err := os.Create(path.Join(outDir, fileNamePrefix+"-diffraction-peaks.csv"))
```
Comment (michaelfedell) [inline core/export/exporter.go:247]: does this file need to be closed after all the writing??

**Review by michaelfedell — COMMENTED** [review]
I didn't see anything glaring in this PR, but I'm not as intimate with the intent of the diffraction peak stuff.

So tentative LGTM from me 👍

File: api/endpoints/middlewareLogger.go [hunk 2]
```diff
@@ -173,12 +173,12 @@ func (h *LoggerMiddleware) Middleware(next http.Handler) http.Handler {
...
 				}
 
-				if userObj.Config.DataCollection != "" {
+				if userObj.Config.DataCollection != "unknown" && userObj.Config.DataCollection != "false" {
 					track = true
 					h.Notifications.SetTrack(requestingUser.UserID, true)
 				}
 			}
-			if track {
+			if track && len(contType) > 0 && contType != "application/octet-stream" {
```
[the remark this answers is not in the document]
Comment (buggtb) [reply api/endpoints/middlewareLogger.go:181]: it wasn't conditional until a little while ago, but before we did rely on the one above and that started this mess, so I think its better to have our own should the one above change

File: core/export/exporter.go (hunk 1 above)
Comment (buggtb) [reply core/export/exporter.go:247]: good catch this is Ryan's code I'll add it in

**Review by michaelfedell — APPROVED** [review]

**Pull request merged**